WordPress/Classes/Utility/Media/UIImageView+ImageDownloader.swift

```diff
@@ -0,0 +1,57 @@
+import Foundation
+import UIKit
+import Gifu
+
+extension UIImageView {
+    @MainActor
+    var wp: ImageViewExtensions { ImageViewExtensions(imageView: self) }
+}
+
+@MainActor
+struct ImageViewExtensions {
```

Comment:
I think you can take this a step further to support `ImageView` too. Something like this

```swift
protocol AsyncImageView {
  func controller(_: ImageViewController, state: State)
}

struct ImageViewExtensions<View: AsyncImageView> {
  let view: View
  ...
}

extension ImageView: AsyncImageView { ... }
extension UIImageView: AsyncImageView { ... }
```

Reply:
I wanted to avoid introducing a protocol to avoid coupling `ImageViewController` with the UI. It can now be used in SwiftUI and it can be used in other `UIImageView` extensions if needed. I'd keep it as is for now because there is only one method so there is no significant difference whether it's a protocol or a closure.